anchor/anchorRecord.go

```diff
@@ -17,22 +17,32 @@ import (
 	"github.com/FactomProject/factomd/common/primitives"
 )
 
-//AnchorRecord is used to construct anchor chain
+// CurrentAnchorVersion is the current anchor version
+const CurrentAnchorVersion int = 1
```

Comment:
We're actually using anchor record V2. You can tell because v1 has signature in the content field, and v2 has it in the external ids. The only caveat is that whoever wrote the bitcoin anchor writer did not update the version in their json. As you can see in this recent bitcoin entry:
https://explorer.factom.com/chains/df3ade9eec4b08d5379cc64270c30ea7315d8a8a1a69efe2b98a60ecdd69e604/entries/002377b0381f4eb48df0deb4025a6d98323434d60ff999fb29a27939e9177a04

Ethereum anchors use to correct version 2 in the json though:
https://explorer.factom.com/chains/6e4540d08d5ac6a1a394e982fb6a2ab8b516ee751c37420055141b94fe070bfe/entries/45af639e22d2c557f6f01fa6492acd45667ae63f4d9dcc97e18b7bba8e600e0f

Good catch though. I'll make a ticket to change that part of the bitcoin anchoring code.

I don't know how much value the constant provides here though, since we'll always need to be able to unmarshal v1 and v2 when nodes are syncing

Reply:
Ah, I thought V2 indicated a second version of the function, rather than a different version of AnchorRecord. I added clarifying comments about v1 vs v2. I also modified the anchor tests to use v2 where applicable. Except for TestValidateAnchorRecordV2, this test fails if I move the anchor string to use version 2 instead of version 1, so I left it. I'm not yet sure why it fails. I also removed the const Version I put in